File: CapStone_Forshop/app/src/main/java/org/pytorch/demo/objectdetection/GridOverlayView.java
package org.pytorch.demo.objectdetection;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.DashPathEffect;
import android.graphics.Paint;
import android.util.AttributeSet;
import android.view.View;

public class GridOverlayView extends View {
    private Paint mGridPaint;

    public GridOverlayView(Context context) {
        super(context);
        init();
    }

    public GridOverlayView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    private void init() {
        mGridPaint = new Paint();
        mGridPaint.setColor(Color.WHITE);
        mGridPaint.setStrokeWidth(2); // 선의 굵기를 2로 설정
        mGridPaint.setStyle(Paint.Style.STROKE);
        // 점선 대신 실선으로 설정
        mGridPaint.setPathEffect(null);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);

        int width = canvas.getWidth();
        int height = canvas.getHeight();

        int rowCount = 3;
        int columnCount = 3;

        int rowHeight = height / rowCount;
        int columnWidth = width / columnCount;

        // 가로선 그리기
        for (int i = 1; i < rowCount; i++) {
            int y = i * rowHeight;
            canvas.drawLine(0, y, width, y, mGridPaint);
        }

        // 세로선 그리기
        for (int i = 1; i < columnCount; i++) {
            int x = i * columnWidth;
            canvas.drawLine(x, 0, x, height, mGridPaint);
        }
    }
}

File: Android/app/src/main/java/com/example/a4shop/GridOverlayView.kt
package com.example.a4shop

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.util.AttributeSet
import android.view.View

class GridOverlayView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : View(context, attrs, defStyleAttr) {

    private val paint = Paint().apply {
        color = Color.WHITE
        style = Paint.Style.STROKE
        strokeWidth = 4f
    }

    var showGrid = false
        set(value) {
            field = value
            invalidate()
        }

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)

        if (!showGrid) return

        // 카메라 프리뷰의 크기를 가져와써
        val parentWidth = (parent as View).width.toFloat()
        val parentHeight = (parent as View).height.toFloat()

        val numCells = 3

        val cellWidth = parentWidth / numCells
        val cellHeight = parentHeight / numCells

        // 격자 그리기
        for (i in 1 until numCells) {
            val startX = cellWidth * i
            canvas.drawLine(startX, 0f, startX, parentHeight, paint)
        }

        for (i in 1 until numCells) {
            val startY = cellHeight * i
            canvas.drawLine(0f, startY, parentWidth, startY, paint)
        }
    }
}
